### PR TITLE
Add regression test for ExpressionTree issue

### PR DESCRIPTION
**Pull request by agocke**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems?id=481125

### CODE REVIEW

**Comment by agocke** [comment]
ping @dotnet/roslyn-compiler for review

**Comment by agocke** [comment]
ping @dotnet/roslyn-compiler for review

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs [hunk 1]
```diff
@@ -31,6 +31,48 @@ public static IMethodSymbol FindLocalFunction(this CompilationVerifier verifier,
...
+
+    public void Dispose() => throw new NotImplementedException();
+}
+
+public class E
+{
+    public int Id;
+}
+", references: new[] { LinqAssemblyRef });
+            CompileAndVerify(comp);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs]: >CompileAndVerify(comp); [](start = 12, length = 23)

Would it make sense to run the compiled assembly and have it print something to confirm that it behaves as expected?

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1)

**Pull request merged**